apps/examples/src/examples/context-toolbar/ContextToolbar.tsx

```diff
@@ -0,0 +1,130 @@
+	const editor = useEditor()
+	const showToolbar = editor.isIn('select.idle')
+	if (!showToolbar) return null
+	const selectionRotatedPageBounds = editor.getSelectionRotatedPageBounds()
+	if (!selectionRotatedPageBounds) return null
+
+	// [2]
+	const styles = new SharedStyleMap(editor.getSharedStyles())
+	const size = styles.get(DefaultSizeStyle)
+	if (!size) return null
+	const currentSize = size.type === 'shared' ? size.value : undefined
+
+	const pageCoordinates = Vec.Sub(
+		editor.pageToScreen(selectionRotatedPageBounds.point),
+		editor.getViewportScreenBounds()
+	)
+
+	return (
+		<div
+			style={{
+				position: 'absolute',
+				top: Math.max(16, pageCoordinates.y - 48),
+				left: Math.max(16, pageCoordinates.x),
+				pointerEvents: 'all',
+				// [3]
+				width: selectionRotatedPageBounds.width,
+			}}
+			// [4]
+			onPointerDown={(e) => e.stopPropagation()}
+		>
+			<div
+				style={{
+					display: 'flex',
+					justifyContent: 'center',
+					alignItems: 'center',
+				}}
+			>
+				<div
+					style={{
+						borderRadius: 8,
+						display: 'flex',
+						boxShadow: '0 0 0 1px rgba(0,0,0,0.1), 0 4px 8px rgba(0,0,0,0.1)',
+						background: 'var(--color-panel)',
+						width: 'fit-content',
+						alignItems: 'center',
+					}}
+				>
+					{STYLES.size.map((v) => {
+						const { value } = v
+						const isActive = value === currentSize
+						return (
+							<div
+								key={value}
+								style={{
+									display: 'flex',
+									alignItems: 'center',
+									justifyContent: 'center',
+									height: 32,
+									width: 32,
+									background: isActive ? 'var(--color-muted-2)' : 'transparent',
+								}}
+								onClick={() =>
+									editor.setStyleForSelectedShapes(DefaultSizeStyle, value, { squashing: false })
+								}
+							>
+								<Icon icon={v.icon} />
+							</div>
+						)
+					})}
+				</div>
+			</div>
+		</div>
+	)
+})
+
+const components: TLEditorComponents = {
+	InFrontOfTheCanvas: ContextToolbarComponent,
+}
+
+export default function ContextToolbar() {
+	return (
+		<div className="tldraw__editor">
+			<Tldraw persistenceKey="things-on-the-canvas-example" components={components} />
+		</div>
+	)
+}
+
+/*
+This example shows how you can impelemnt a context toolbar that appears when you select shapes.
+It's using the `InFrontOfTheCanvas` component to render the toolbar. This allows use to render
+the toolbar on top of the shapes, but behind the existing UI. The toolbar is only rendered when
+we are in the `select.idle` state (so we are not rotating, moving, resizing, drawing,...).
+You can also check the `Things on the canvas` example for more info on how to use `TLEditorComponents`.
+
+[1]
+This is our context toolbar. It's positioned absolutely on top of the selected shapes.
+
+[2]
+Different shapes support different style properties and this is how we get the styles that are 
+supported by all selected shapes. If some of the selected shapes don't support the `DefaultSizeStyle`
```

Comment:
I think this is the behaviour, isn't it?
```suggestion
supported by all selected shapes. If none of the selected shapes supported the `DefaultSizeStyle`
```